<a href="https://colab.research.google.com/github/basharataly/nyu-python-public-policy/blob/master/HW3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [0]:
import pandas as pd
from google.colab import drive

In [0]:
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)  

In [16]:
drive.mount('/content/drive')

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3aietf%3awg%3aoauth%3a2.0%3aoob&response_type=code&scope=email%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdocs.test%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.photos.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fpeopleapi.readonly

Enter your authorization code:
··········
Mounted at /content/drive


In [0]:
df = pd.read_csv('/content/drive/My Drive/Data for Python/cleaned_311_data_hw3.csv', header='infer')

In [18]:
new  = df.groupby(['complaint_type', 'descriptor']).size().reset_index(name='count')
new


complaint_type                           descriptor  count
0             Bridge Condition                        American Flag      6
1             Bridge Condition         Decorative Necklace Lighting      5
2             Bridge Condition       Dirt, Debris, Litter Complaint     74
3             Bridge Condition            Fallen Debris from Bridge     23
4             Bridge Condition                                Fence     17
5             Bridge Condition                    Graffiti - Bridge     60
6             Bridge Condition                  Guard Rail - Bridge     20
7             Bridge Condition                              Pothole    134
8             Bridge Condition                       Rough Pavement     40
9             Bridge Condition                     Snow on Overpass     13
10        Broken Parking Meter                           Bent/Loose     64
11        Broken Parking Meter        Coin or Card Did Not Register   3833
12        Broken Parking Meter           Credit Card Stuck in Meter    373
13        Broken Parking Meter                             Graffiti     83
14        Broken Parking Meter                        Missing/Stump    122
15        Broken Parking Meter                           No Receipt   5206
16        Broken Parking Meter                         Out of Order   3787
17        Broken Parking Meter          Parking Card Stuck in Meter    139
18        Broken Parking Meter          Relocation of Parking Meter      9
19        Broken Parking Meter             Timer Defect - Fast/Fail    594
20  Bus Stop Shelter Complaint                         Broken Glass    391
21  Bus Stop Shelter Complaint                        Damaged Bench     21
22  Bus Stop Shelter Complaint             Damaged Leg or Pole Bent     11
23  Bus Stop Shelter Complaint                        Damaged Other    109
24  Bus Stop Shelter Complaint              Damaged or Leaking Roof      2
25  Bus Stop Shelter Complaint            Damaged or Missing Ad Box     22
26  Bus Stop Shelter Complaint                       Dirty/Graffiti    187
27  Bus Stop Shelter Complaint                             Lighting    140
28  Bus Stop Shelter Complaint            Sign Missing or Defective     93
29  Bus Stop Shelter Complaint               Snow Removal Requested     14
30                    Graffiti                             Graffiti  24707
31                    Graffiti          Police Report Not Requested    168
32                    Graffiti              Police Report Requested    111
33           Highway Condition                              Cave-in    236
34           Highway Condition                     Concrete Barrier      1
35           Highway Condition                 Crash Cushion Defect     38
36           Highway Condition                          Dead Animal    300
37           Highway Condition                   Graffiti - Highway    310
38           Highway Condition                 Guard Rail - Highway      8
39           Highway Condition                           High Grass    147
40           Highway Condition                        Highway Fence     61
41           Highway Condition                               Litter    572
42           Highway Condition                          Loose Plate     89
43           Highway Condition                 Painted Line/Marking    184
44           Highway Condition                    Pothole - Highway   1505
45           Highway Condition                 Sign Blocked by Tree     16
46           Highway Condition                      Unsafe Worksite     76
47     Maintenance or Facility                         Broken Glass    210
48     Maintenance or Facility                    Garbage or Litter   3500
49     Maintenance or Facility                Graffiti or Vandalism    577
50     Maintenance or Facility                          Grass/Weeds   2084
51     Maintenance or Facility                   Hours of Operation    697
52     Maintenance or Facility           

In [0]:
def sf(row):
  if 'Graffiti'  in row.complaint_type:
    return True 
  elif 'Graffiti' in row.descriptor:
    return True
  else:
    return False

In [24]:
type(sf)

function

In [0]:
new['graffiti'] = new.apply(sf, axis=1)

In [31]:
new

complaint_type                           descriptor  count  \
0             Bridge Condition                        American Flag      6   
1             Bridge Condition         Decorative Necklace Lighting      5   
2             Bridge Condition       Dirt, Debris, Litter Complaint     74   
3             Bridge Condition            Fallen Debris from Bridge     23   
4             Bridge Condition                                Fence     17   
5             Bridge Condition                    Graffiti - Bridge     60   
6             Bridge Condition                  Guard Rail - Bridge     20   
7             Bridge Condition                              Pothole    134   
8             Bridge Condition                       Rough Pavement     40   
9             Bridge Condition                     Snow on Overpass     13   
10        Broken Parking Meter                           Bent/Loose     64   
11        Broken Parking Meter        Coin or Card Did Not Register   3833   
12        Broken Parking Meter           Credit Card Stuck in Meter    373   
13        Broken Parking Meter                             Graffiti     83   
14        Broken Parking Meter                        Missing/Stump    122   
15        Broken Parking Meter                           No Receipt   5206   
16        Broken Parking Meter                         Out of Order   3787   
17        Broken Parking Meter          Parking Card Stuck in Meter    139   
18        Broken Parking Meter          Relocation of Parking Meter      9   
19        Broken Parking Meter             Timer Defect - Fast/Fail    594   
20  Bus Stop Shelter Complaint                         Broken Glass    391   
21  Bus Stop Shelter Complaint                        Damaged Bench     21   
22  Bus Stop Shelter Complaint             Damaged Leg or Pole Bent     11   
23  Bus Stop Shelter Complaint                        Damaged Other    109   
24  Bus Stop Shelter Complaint              Damaged or Leaking Roof      2   
25  Bus Stop Shelter Complaint            Damaged or Missing Ad Box     22   
26  Bus Stop Shelter Complaint                       Dirty/Graffiti    187   
27  Bus Stop Shelter Complaint                             Lighting    140   
28  Bus Stop Shelter Complaint            Sign Missing or Defective     93   
29  Bus Stop Shelter Complaint               Snow Removal Requested     14   
30                    Graffiti                             Graffiti  24707   
31                    Graffiti          Police Report Not Requested    168   
32                    Graffiti              Police Report Requested    111   
33           Highway Condition                              Cave-in    236   
34           Highway Condition                     Concrete Barrier      1   
35           Highway Condition                 Crash Cushion Defect     38   
36           Highway Condition                          Dead Animal    300   
37           Highway Condition                   Graffiti - Highway    310   
38           Highway Condition                 Guard Rail - Highway      8   
39           Highway Condition                           High Grass    147   
40           Highway Condition                        Highway Fence     61   
41           Highway Condition                               Litter    572   
42           Highway Condition                          Loose Plate     89   
43           Highway Condition                 Painted Line/Marking    184   
44           Highway Condition                    Pothole - Highway   1505   
45           Highway Condition                 Sign Blocked by Tree     16   
46           Highway Condition                      Unsafe Worksite     76   
47     Maintenance or Facility                         Broken Glass    210   
48     Maintenance or Facility                    Garbage or Litter   3500   
49     Maintenance or Facility                Graffiti or Vandalism    577   
50     Maintenance or Facility        

In [32]:
df['graffiti_flag']=df.apply(sf, axis=1)
df.head(5)

Unnamed: 0  Unnamed: 0.1  unique_key            created_date  \
0         238           238    39887282  08/01/2018 01:42:01 AM   
1         353           353    39889407  08/01/2018 03:20:07 AM   
2         403           403    39893067  08/01/2018 04:19:15 AM   
3         655           655    39890578  08/01/2018 07:32:33 AM   
4         658           658    39896325  08/01/2018 07:33:12 AM   

              closed_date agency                         agency_name  \
0  08/14/2018 02:42:12 PM    DPR  Department of Parks and Recreation   
1  08/23/2018 11:15:00 AM    DPR  Department of Parks and Recreation   
2  08/02/2018 08:19:55 AM    DPR  Department of Parks and Recreation   
3  08/16/2018 10:28:38 AM    DPR  Department of Parks and Recreation   
4  09/04/2018 05:38:02 PM    DPR  Department of Parks and Recreation   

            complaint_type            descriptor location_type  incident_zip  \
0  Maintenance or Facility  Structure - Outdoors          Park       10003.0   
1  Maintenance or Facility     Garbage or Litter          Park       11434.0   
2  Maintenance or Facility    Hours of Operation          Park       11206.0   
3  Maintenance or Facility     Garbage or Litter          Park       11691.0   
4  Maintenance or Facility     Garbage or Litter          Park       11355.0   

  incident_address street_name cross_street_1 cross_street_2  \
0              NaN         NaN            NaN            NaN   
1              NaN         NaN            NaN            NaN   
2              NaN         NaN            NaN            NaN   
3              NaN         NaN            NaN            NaN   
4              NaN         NaN            NaN            NaN   

  intersection_street_1 intersection_street_2  address_type          city  \
0                   NaN                   NaN           NaN      NEW YORK   
1     BAISLEY BOULEVARD     MERRICK BOULEVARD  INTERSECTION       JAMAICA   
2                   NaN                   NaN           NaN      BROOKLYN   
3                   NaN                   NaN           NaN  FAR ROCKAWAY   
4                   NaN                   NaN           NaN      FLUSHING   

  landmark facility_type  status                due_date  \
0      NaN           NaN  Closed  08/15/2018 01:42:01 AM   
1      NaN           NaN  Closed  08/04/2018 03:20:07 AM   
2      NaN           NaN  Closed  08/04/2018 04:19:15 AM   
3      NaN           NaN  Closed  08/04/2018 07:32:33 AM   
4      NaN           NaN  Closed  08/04/2018 07:33:12 AM   

                              resolution_description  \
0  The Department of Parks and Recreation has com...   
1  The Service Request submitted did not have suf...   
2  The Department of Parks and Recreation has com...   
3  The Department of Parks and Recreation has com...   
4  The condition was inspected and it was determi...   

  resolution_action_updated_date community_board  bbl      borough  \
0         08/14/2018 02:42:12 PM   0 Unspecified  NaN  Unspecified   
1         08/23/2018 11:15:00 AM       12 QUEENS  NaN       QUEENS   
2         08/02/2018 08:19:55 AM     01 BROOKLYN  NaN     BROOKLYN   
3         08/16/2018 10:28:38 AM       14 QUEENS  NaN       QUEENS   
4         09/04/2018 05:38:02 PM       07 QUEENS  NaN       QUEENS   

   x_coordinate_(state_plane)  y_coordinate_(state_plane)  \
0                         NaN                         NaN   
1                   1048111.0                    188479.0   
2                         NaN                         NaN   
3                         NaN                         NaN   
4                         NaN                         NaN   

  open_data_channel_type     park_facility_name park_borough  vehicle_type  \
0                  PHONE             First Park  Unspecified           NaN   
1                  PHONE            Unspecified       QUEENS           NaN   
2                 MOBILE         Ten Eyck Plaza     BROOKLYN           NaN   
3                  PHONE  Westbourne Pla

In [0]:
df_graffiti = df.groupby(['graffiti_flag', 'community_board']).size().reset_index(name='count')

In [34]:
df_graffiti.max()

graffiti_flag                           True
community_board    Unspecified STATEN ISLAND
count                                   8044
dtype: object

Unspecified STATEN ISLAND has the highest count of graffiti with a count of 8044.
